python/semantic_kernel/kernel.py

```diff
@@ -102,24 +104,56 @@ def prompt_template_engine(self) -> PromptTemplatingEngine:
         return self._prompt_template_engine
 
     @property
-    def plugins(self) -> ReadOnlyPluginCollectionBase:
-        return self._plugin_collection.read_only_plugin_collection
+    def plugins(self) -> KernelPluginCollection:
+        return self._plugins
+
+    @plugins.setter
+    def plugins(self, value: KernelPluginCollection) -> None:
+        self._plugins = value
+
+    def add_plugin_to_collection(self, plugin_name: str, functions: List[KernelFunctionBase]) -> None:
```

Comment:
Would it make sense to also allow to add an existing `KernelPlugin` instance?

Reply:
I have added this in.